src/Microsoft.Health.SqlServer/DefaultSqlConnectionFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Data.SqlClient;
+using EnsureThat;
+using Microsoft.Health.SqlServer.Configs;
+
+namespace Microsoft.Health.SqlServer
+{
+    /// <summary>
+    /// Default Sql Connection Factory is responsible to handle Sql connections that can be made purely based on connection string.
+    /// Connection string containing user name and password, or integrated auth are perfect examples for this.
+    /// </summary>
+    public class DefaultSqlConnectionFactory : ISqlConnectionFactory
```

Comment:
Where are the other implementations? We should keep all the implementations here so that when Fhir needs it when they move to AKS they will have it here easily

Reply:
I kept ManagedIdentitySqlConnectionFactory in DICOM thinking that there may be other ways to fetch access token for different services. I can move it here and any service that needs a different implementation can override it.

Reply:
Moved ManagedIdentitySqlConnection to shared components solutions. DefaultSqlConnection can be used for SqlUserNameAndPassword connection type and WindowsIntegratedAuth connection type as they both just rely on the connection string.